shelley/design-spec/delegation_design_spec.tex

```diff
@@ -852,106 +804,83 @@ \subsubsection{Reward Account Address}
 
-Bootstrap addresses continue to exist in Shelley, but their
+Byron addresses continue to exist in Shelley, but their
 interpretation is changed subtly and their use is disincentivised.
-Their interpretation is changed from having stake rights but with
+Their interpretation is changed from having stake rights with
 forced delegation, to having no stake rights whatsoever. Their use is
-disincentivised since owners have the option to move their funds into
-the new base or pointer addresses that have stake rights which can be
-exercised to receive staking rewards.
+disincentivised, since owners have the option to move their funds into
+the new base or pointer addresses that have stake rights, which can be
+exercised to receive rewards.
 
-It is worth noting that initially bootstrap addresses and enterprise addresses
+It is worth noting that initially, Byron addresses and enterprise addresses
 have essentially identical behaviour. This might change in the future, if new
 features are added to enterprise addresses.
 
-\subsubsection{Script Address}
-\label{script-address}
-
-Byron does not support scripts. Scripts are \emph{not} included in the
-Shelley design described in this document.
-
-Scripts will however be introduced in future as an extension of the
-Shelley design and we will briefly describe the interaction between the
-future script addresses and stake rights.
-
-Script addresses will have the form
-\[
-\hash{vs} \mathbin{||} \beta
-\]
-where \(\hash{vs}\) is a cryptographic hash of the validation script.
-
-The staking object \(\beta\) can take the same three forms as normal
-pubkey addresses: base, pointer and none. The use of stake rights with
-script addresses is otherwise identical to that of normal addresses.
-
-The use of script addresses with no stake rights has the same behaviour
-as normal addresses with no stake rights, but has different motivating
-use cases. For many scripts it is hard to determine to whom the funds
-actually belong and in such cases the simplest solution is to use an
-address with no stake rights whatsoever. For this reason we do not call
-such addresses enterprise addresses.
-
 \subsubsection{HD Wallet Structure in Shelley}
 \label{hd-wallet-structure-in-shelley}
 
-All the Shelley address formats (other than script addresses) support
-hierarchical deterministic wallets, as per BIP-32 \citep{bip32}.
+Shelley addresses with a verification key hash as payment credential
+support hierarchical deterministic wallets, as per BIP-32~\citep{bip32}.
 
-In particular this kind of wallet scheme allows implementations that can
+For multi-sig value addresses, we can use a slight generalisation of
+BIP-45~\citep{bip45}\todo{Lay out the necessary generalisation of BIP-45}.
+
+In particular, this kind of wallet scheme allows implementations that can
 do wallet restoration from seed in time that is better than linear in
 the total number of addresses in the blockchain. For details, see
 \cref{wallet-recovery-process}.
```

Comment:
What does better than linear actually mean here?  Strictly sub-linear is often used as a term, but you might mean asymptotically linear (but often better)?  If it's log or sqrt, say so...